File: packages/shared-protocol/src/ts/doc-review.ts
/**
 * 文档风险合规审核契约 — TS 镜像（snake_case，与 FastAPI 响应一致，参照 dspark.ts 先例）。
 * Spec: docs/superpowers/specs/2026-08-04-doc-risk-review-design.md
 */

export type DocFormat = 'pdf' | 'docx' | 'txt' | 'md';
export type DocCategory =
  | 'contract'
  | 'internal_policy'
  | 'announcement'
  | 'bidding'
  | 'other';
export type DocRiskType = 'compliance' | 'legal' | 'data_security' | 'financial';
export type DocRiskLevel = 'low' | 'medium' | 'high' | 'critical';
export type DocRunStatus = 'queued' | 'classifying' | 'analyzing' | 'done' | 'failed';

export interface DocBlock {
  block_id: string;
  text: string;
  start: number;
  end: number;
}

export interface DocPage {
  page_no: number;
  blocks: DocBlock[];
}

export interface DocPosition {
  page_no: number;
  block_id: string;
  start: number;
  end: number;
}

export interface DocFinding {
  finding_id: string;
  risk_type: DocRiskType;
  risk_level: DocRiskLevel;
  title: string;
  description: string;
  suggestion: string;
  rule_ref: string | null;
  evidence_text: string;
  positions: DocPosition[];
}

export interface DocSummary {
  doc_id: string;
  file_name: string;
  format: DocFormat;
  page_count: number;
  status: DocRunStatus | 'none';
  overall_risk_level: DocRiskLevel | null;
  created_at: string;
}

export interface DocDetail extends DocSummary {
  file_path: string;
  doc_category: DocCategory | null;
  risk_types: DocRiskType[];
  summary: string | null;
  pages: DocPage[];
  findings: DocFinding[];
}

File: packages/shared-protocol/src/protocol/doc_review.py
"""文档风险合规审核契约 (Pydantic 镜像, 字段与 ts/doc-review.ts 一致)."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

DocFormat = Literal["pdf", "docx", "doc", "txt", "md", "html", "xlsx", "pptx"]
DocCategory = Literal["contract", "internal_policy", "announcement", "bidding", "other"]
DocRiskType = Literal["compliance", "legal", "data_security", "financial"]
DocRiskLevel = Literal["low", "medium", "high", "critical"]
DocRunStatus = Literal["queued", "classifying", "analyzing", "done", "failed"]


class DocBlock(BaseModel):
    block_id: str
    text: str
    start: int = Field(ge=0)
    end: int = Field(ge=0)


class DocPage(BaseModel):
    page_no: int = Field(ge=1)
    blocks: list[DocBlock]


class DocPosition(BaseModel):
    page_no: int = Field(ge=1)
    block_id: str
    start: int = Field(ge=0)
    end: int = Field(ge=0)


class DocKbRef(BaseModel):
    """知识库/案例库引用（grep 式匹配，解释"为什么有风险"）。"""

    source: str
    heading: str
    excerpt: str = ""
    matched_terms: list[str] = Field(default_factory=list)


class DocFinding(BaseModel):
    finding_id: str
    risk_type: DocRiskType
    risk_level: DocRiskLevel
    title: str
    description: str = ""
    suggestion: str = ""
    rule_ref: str | None = None
    evidence_text: str = ""
    positions: list[DocPosition] = Field(default_factory=list)
    kb_refs: list[DocKbRef] = Field(default_factory=list)


class DocSummary(BaseModel):
    doc_id: str
    file_name: str
    format: DocFormat
    page_count: int = Field(ge=1)
    status: DocRunStatus | Literal["none"]
    overall_risk_level: DocRiskLevel | None = None
    created_at: str


class DocDetail(DocSummary):
    file_path: str
    doc_category: DocCategory | None = None
    risk_types: list[DocRiskType] = Field(default_factory=list)
    summary: str | None = None
    pages: list[DocPage] = Field(default_factory=list)
    findings: list[DocFinding] = Field(default_factory=list)
